cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerLoggingConfigurationDiff.java

```diff
@@ -96,6 +95,80 @@ public int getDiffSize() {
         return updatedCE;
     }
 
+    private static Map<String, String> readLog4jConfig(String config) {
+
+        Map<String, String> parsed = new LinkedHashMap<>();
+        Map<String, String> env = new HashMap<>();
+        BufferedReader reader = new BufferedReader(new StringReader(config));
+        String line;
+        try {
+            while ((line = reader.readLine()) != null) {
+                // skip comments
+                if (line.startsWith("#")) continue;
+
+                // ignore empty lines
+                line = line.trim();
+                if (line.length() == 0) continue;
+
+                // everything that does not start with 'log4j.' is a variable definition
+                if (!line.startsWith("log4j.")) {
+                    int p = line.indexOf("=");
+                    if (p >= 0) {
+                        env.put(line.substring(0, p).trim(), line.substring(p + 1).trim());
+                    } else {
+                        env.put(line.trim(), "");
+                    }
+                    log.debug("Treating the line as ENV var declaration: {}", line);
+                    continue;
+                }
+
+                // we ignore appenders (log4j.appender.*)
+                // and only handle loggers (log4j.logger.*)
+                if (line.startsWith("log4j.logger.")) {
+                    int p = line.indexOf("=", 13);
+                    if (p == -1) {
+                        log.debug("Skipping log4j.logger.* declaration without level: {}", line);
+                        continue;
+                    }
+                    String name = line.substring(13, p).trim();
+                    String value = line.substring(p + 1).trim();
+
+                    value = expandVars(value, env);
+                    parsed.put(name, value);
+
+                } else if (line.startsWith("log4j.rootLogger=")) {
+                    parsed.put("root", expandVars(line.substring(17).trim(), env));
+
+                } else {
+                    log.debug("Skipping log4j line: {}", line);
+                }
+            }
+        } catch (Exception e) {
+            log.error("Failed to parse logging configuration: " + config, e);
+            return Collections.emptyMap();
+        }
+        return parsed;
+    }
+
+    private static String expandVars(String value, Map<String, String> env) {
+        StringBuilder sb = new StringBuilder();
+        int e = -1;
+        int b;
```

Comment:
Likewise, why we are using single letter variable here? The rest of the Strimzi codebase is pretty consistently using descriptive names for variables like these.

Reply:
I agree with Tom. It's hard to read code with one letter variables.

Reply:
I can see the majority opinion is against me on this one :)
I'll call these startIdx and endIdx then.